packages/proxy/lib/http/index.ts

```diff
@@ -331,12 +331,14 @@ export class Http {
 
     const onError = (error: Error): Promise<void> => {
       ctx.error = error
-      if (ctx.req.browserPreRequest) {
+      if (ctx.req.browserPreRequest && !ctx.req.browserPreRequest.errorHandled) {
+        ctx.req.browserPreRequest.errorHandled = true
```

Comment:
Do we need a test for the `errorHandled` logic?

Reply:
Added: [`54321fc` (#27892)](https://github.com/cypress-io/cypress/pull/27892/commits/54321fc0b0c5f582d12a43dd32dd0b9b508945b8)